metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py

```diff
@@ -0,0 +1,204 @@
+import logging
+from typing import Optional
+
+import pydantic
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from snowflake.connector.network import (
+    DEFAULT_AUTHENTICATOR,
+    EXTERNAL_BROWSER_AUTHENTICATOR,
+    KEY_PAIR_AUTHENTICATOR,
+)
+
+from datahub.configuration.common import AllowDenyPattern, ConfigModel
+from datahub.configuration.time_window_config import BaseTimeWindowConfig
+from datahub.ingestion.source.sql.sql_common import (
+    SQLAlchemyConfig,
+    make_sqlalchemy_uri,
+)
+from datahub.utilities.config_clean import (
+    remove_protocol,
+    remove_suffix,
+    remove_trailing_slashes,
+)
+
+APPLICATION_NAME = "acryl_datahub"
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class SnowflakeSetupConfig(ConfigModel):
```

Comment:
please add comment to the model. also think we should consider renaming the inner fields, or at least nesting inside of provision_role or something that is named to reflect what is happening

Reply:
changed to `provision_role`